### PR TITLE
User story timer

### PR DESCRIPTION
**Pull request by reyk-zepper**
- implement the possibility to record the working-time for a project
- display the working time in details view
- minimal styling with styled components

### CODE REVIEW

File: components/EditForm/index.js [hunk 1]
```diff
@@ -98,11 +99,12 @@ export default function EditForm({ project }) {
   };
 
   if (project === undefined) {
-    return <h2> there is no prject with this id: {id}</h2>;
+    return <h2> there is no project with this id: {id}</h2>;
   }
 
   return (
     <div>
```
Comment (PReichetanz) [inline components/EditForm/index.js:106]: You could use a fragment here instead and would thus get rid of an unnecessary `div`:
```suggestion
    <>
```

File: helper/formatTime.js
Comment (PReichetanz) [inline helper/formatTime.js:1]: I guess this would be a good helper function to write a unit test for ;)

File: hooks/useStore.js [hunk 2]
```diff
@@ -30,6 +30,22 @@ const config = (set) => {
       });
     },
 
+    updateTimer: (editProject) => {
+      set((draft) => {
+        const draftProjectArr = draft.projects.map((project) => {
```
Comment (PReichetanz) [inline hooks/useStore.js:35]: Does this mean `draftProjectArray` or `draftProjectArrow`? I think that there is no need to abbreviate here and instead be specific about what's going on.

File: pages/edit/[id].js [hunk 3]
```diff
@@ -28,6 +28,10 @@ export default function EditPage() {
   if (project === " " || project === undefined) {
     return <h2>loading...</h2>;
   } else {
-    return <EditForm project={project} />;
+    return (
+      <>
+        <EditForm project={project} />
+      </>
```
Comment (PReichetanz) [inline pages/edit/[id].js:34]: Why did you add the fragments here? I don't see any need for them.

**Review by PReichetanz — COMMENTED** [review]
Nice work! Left some comments to discuss, but no big issues.

File: helper/formatTime.js
Comment (reyk-zepper) [reply helper/formatTime.js:1]: thx, i have add a format test for it

File: pages/edit/[id].js (hunk 3 above)
Comment (reyk-zepper) [reply pages/edit/[id].js:34]: yapp. you're right...it's a left over...i have removed it

File: components/EditForm/index.js (hunk 1 above)
Comment (reyk-zepper) [reply components/EditForm/index.js:106]: thx for the hint...i done it

File: hooks/useStore.js (hunk 2 above)
Comment (reyk-zepper) [reply hooks/useStore.js:35]: it means draftProjectArray, i map over the projects and store the result in the draftProjectArray. for me it was clear cause the map function, that results an array. but i have changed it for better readability.

**Pull request merged**